pkg/apis/kube/namespaces.go

```
@@ -69,3 +71,40 @@ func (kube *Kube) ApplyNamespace(name string, annotations map[string]string, lab
 
 	return err
 }
+
+// NamespaceWatcher Watcher to wait for namespace to be created
+type NamespaceWatcher interface {
+	WaitFor(namespace string) error
+}
+
+// NamespaceWatcherImpl Implementation of watcher
+type NamespaceWatcherImpl struct {
+	client kubernetes.Interface
+}
+
+// NewNamespaceWatcherImpl Constructor
+func NewNamespaceWatcherImpl(client kubernetes.Interface) NamespaceWatcherImpl {
+	return NamespaceWatcherImpl{
+		client,
+	}
+}
+
+// WaitFor Waits for namespace to appear
+func (watcher NamespaceWatcherImpl) WaitFor(namespace string) error {
```

Error is never returned (except nil)

---

Return error now, also on timeouts